data_iterator.go

```diff
@@ -45,140 +50,198 @@ func (d *DataIterator) Run(tables []*TableSchema) {
 		d.StateTracker.MarkTableAsCompleted(table.String())
 	}
 
-	for table, maxPaginationKey := range tablesWithData {
+	batchQueue := make(chan *DataIteratorBatch)
```

Comment:
`batchQueue := make(chan *DataIteratorBatch, d.Concurrency)`

Your for loop is going to get blocked a few times if you don't.